processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -184,6 +203,7 @@ protected void scanAndAggregate(
 
     final int aggSize = theAggregators.length;
     final int aggExtra = aggSize % AGG_UNROLL_COUNT;
+    final AtomicInteger currentPosition = new AtomicInteger(0);
```

Comment:
This is not for thread safe. currentPosition will be changed in another method i use AtomicInteger to record the changed value.